pkg/resources/block_storage/resource_volume.go

```diff
@@ -0,0 +1,588 @@
+package block_storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	exoscale "github.com/exoscale/egoscale/v3"
+
+	"github.com/exoscale/terraform-provider-exoscale/pkg/config"
+	providerConfig "github.com/exoscale/terraform-provider-exoscale/pkg/provider/config"
+	"github.com/exoscale/terraform-provider-exoscale/pkg/utils"
+)
+
+const ResourceVolumeDescription = `Manage [Exoscale Block Storage](https://community.exoscale.com/documentation/block-storage/) Volume.
+
+Block Storage offers persistent externally attached volumes for your workloads.
+`
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &ResourceVolume{}
+var _ resource.ResourceWithImportState = &ResourceVolume{}
+
+// ResourceVolume defines the resource implementation.
+type ResourceVolume struct {
+	client *exoscale.Client
+}
+
+// NewResourceVolume creates instance of ResourceVolume.
+func NewResourceVolume() resource.Resource {
+	return &ResourceVolume{}
+}
+
+// ResourceVolumeModel defines the resource data model.
+type ResourceVolumeModel struct {
+	ID             types.String `tfsdk:"id"`
+	Name           types.String `tfsdk:"name"`
+	Size           types.Int64  `tfsdk:"size"`
+	Blocksize      types.Int64  `tfsdk:"blocksize"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+	Labels         types.Map    `tfsdk:"labels"`
```

Comment:
I think I found an issue related to this field, can it be fixed?

When I apply:
```hcl
resource "exoscale_block_storage_volume" "vol1" {
  name = "vol-1"
  size = 11
  zone = "ch-gva-2"
}
```

I get:
```shell
exoscale_block_storage_volume.vol1: Creating...
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to exoscale_block_storage_volume.vol1, provider
│ "provider[\"registry.terraform.io/exoscale/exoscale\"]" produced an unexpected new value: .labels: was null, but now
│ cty.MapValEmpty(cty.String).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

Reply:
it works if I specify `labels = {}`, I don't know if we want to "fix" this so it allows keeping it unset, or we could set the default value to `{}`.

Reply:
Fixed in https://github.com/exoscale/terraform-provider-exoscale/pull/341/commits/9c05a262e4d1176d922d961506d4bd5a3e428884